client/my-sites/plans/plan-overview/plan-remove/index.jsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Dialog from 'components/dialog';
+import { isInGracePeriod } from 'lib/plans';
+import paths from '../../paths';
+
+const PlanRemove = React.createClass( {
+	propTypes: {
+		plan: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired
+	},
+
+	getInitialState() {
+		return {
+			showDialog: false
+		};
+	},
+
+	closeDialog( action ) {
+		this.setState( { showDialog: false } );
+
+		if ( action === 'remove' ) {
+			// TODO: Add action that calls the API and cancels the free trial
```

Comment:
I know this is `TODO` (and may be a temporary oversight), but what do you think about moving this to its own method? You can attach an event handler to a button in the `onClick` property of the button object in render().

The benefits would be that you don't need to check the type of action, and the action wouldn't be triggered by a method named `closeDialog`, which probably shouldn't handle removing a plan.